src/types/models/mod_feedback.rs

```diff
@@ -0,0 +1,157 @@
+use std::cmp::PartialEq;
+use serde::{Serialize, Deserialize};
+use sqlx::{PgConnection,FromRow};
+
+use crate::types::api::ApiError;
+use crate::types::models::mod_version::ModVersion;
+
+#[derive(FromRow)]
+struct ModFeedbackRow {
+    id: i32,
+    reviewer_id: i32,
+    reviewer_name: String,
+    reviewer_admin: bool,
+    feedback_type: FeedbackTypeEnum,
+    feedback: String,
+    decision: bool,
+}
+
+#[derive(Serialize)]
+pub struct ModFeedback {
+    pub score: i32,
+    pub mod_id: String,
+    pub mod_version: String,
+    pub feedback: Vec<ModFeedbackOne>,
+}
+
+#[derive(Serialize)]
+pub struct Reviewer {
+    pub id: i32,
+    pub display_name: String,
+    pub admin: bool,
+}
+
+#[derive(Serialize)]
+pub struct ModFeedbackOne {
+    #[serde(skip_serializing)]
+    pub id: i32,
+    pub reviewer: Reviewer,
+    pub feedback_type: FeedbackTypeEnum,
+    pub feedback: String,
+    pub decision: bool,
+}
+
+#[derive(sqlx::Type, Serialize, Deserialize, Clone, PartialEq)]
+#[sqlx(type_name = "feedback_type")]
+pub enum FeedbackTypeEnum {
+    Positive,
+    Negative,
+    Suggestion,
+    Note
+}
+
+impl ModFeedback {
+    pub async fn get_for_mod_version_id(
+        version: &ModVersion,
+        note_only: bool,
+        pool: &mut PgConnection,
+    ) -> Result<ModFeedback, ApiError> {
+        let result = match sqlx::query_as!(
+            ModFeedbackRow,
+            r#"SELECT mf.id, mf.reviewer_id, dev.display_name AS reviewer_name, dev.admin AS reviewer_admin, mf.type AS "feedback_type: _", mf.feedback, mf.decision
+            FROM mod_feedback mf
+			INNER JOIN developers dev ON dev.id = mf.reviewer_id
+            WHERE mf.mod_version_id = $1"#,
+            version.id
+        )
+        .fetch_all(&mut *pool)
+        .await
+        {
+            Err(e) => {
+                log::error!("{}", e);
+                return Err(ApiError::DbError);
+            }
+            Ok(r) => r,
+        };
+
+        let feedback: Vec<ModFeedbackOne> = result.into_iter().filter_map(|row| {
+            if note_only && row.feedback_type != FeedbackTypeEnum::Note {
+                None
+            } else {
+                Some(ModFeedbackOne {
+                    id: row.id,
+                    reviewer: Reviewer {
+                        id: row.reviewer_id,
+                        display_name: row.reviewer_name,
+                        admin: row.reviewer_admin,
+                    },
+                    feedback_type: row.feedback_type,
+                    feedback: row.feedback,
+                    decision: row.decision,
+                })
+            }
+        }).collect();
+
+        let positive = feedback.iter().filter(|r| r.feedback_type == FeedbackTypeEnum::Positive).count() as i32;
```

Comment:
i think you could do this in sql (good if pagination is added) but might be preferable to have the api return the frequency of all counts instead, and then let the client do the score calculation. not entirely sure

Reply:
i mean we could, but it's only slowing it down by like a millisecond.

Reply:
also i dont think i can do it in sql - i mean i could, but here it's one request opposed to, like, 3

Reply:
plus, this way it's 1 db hit instead of 3